src/main/java/com/github/matidominati/medicalclinic/personInfo/PatientController.java

```diff
@@ -0,0 +1,55 @@
+package com.github.matidominati.medicalclinic.personInfo;
+
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/patients")
+public class PatientController {
+    private List<Patient> patients = new ArrayList<>();
+
+    @GetMapping
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+
+    @GetMapping("/{email}")
+    public Patient getPerson(@PathVariable String email) {
+        return patients.stream()
+                .filter(patient -> patient.getEmail().equals(email))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Pacjent nie istnieje w bazie"));
+
+    }
+
+    @PostMapping
+    public void addPatient(@RequestBody Patient patient) {
+        if (!patients.contains(patient)) {
+            patients.add(patient);
+        } else throw new RuntimeException("Pacjent już istnieje w bazie");
+    }
+
+    @DeleteMapping("/{email}")
+    public void deletePatient(@PathVariable String email) {
+        Patient patientToDelete = getPerson(email);
```

Comment:
metoda powinna nazywac sie getPatient

Reply:
Nie powinnismy uzywac metod w kodzie ktore sa endpointami. Jesli funkcjonalnosc sie powtarza w tym przypadku pobrania pacjenta to powinnismy napisac po prostu metode

getPatient() a nastepnie ta metode wywolac zarowno tutaj jak i bezposrednio w endpoincie, ktory sluzy do zwracania pacjenta.